static/js/components/PostDisplay.js

```diff
@@ -23,30 +27,41 @@ const postTitle = (post: Post) =>
       {post.title}
     </a>
 
-export default class PostDisplay extends React.Component {
+class PostDisplay extends React.Component {
   props: {
     post: Post,
     showChannelLink?: boolean,
-    expanded?: boolean
+    expanded?: boolean,
+    dispatch: Dispatch
   }
 
   showChannelLink = () => {
     const { post, showChannelLink } = this.props
 
-    return showChannelLink && post.channel_name
+    return (showChannelLink && post.channel_name
       ? <Link to={`/channel/${post.channel_name}`}>
           on {post.channel_name}
       </Link>
-      : null
+      : null)
+  }
+
+  toggleUpvote = () => {
+    const { dispatch, post } = this.props
+    dispatch(actions.postUpvotes.post(post.id, !post.upvoted)).then(post => {
+      dispatch(setPostData(post))
+    })
```

Comment:
In MM we typically defined functions like these (click handlers that make an API call) in containers, then passed them down to the components that need them. We don't necessarily need to stick to that - I know we're rendering this component from 2 different places. Maybe @aliceriot could chime in.